lib/game.rb

```diff
@@ -0,0 +1,131 @@
+class Game
+  attr_accessor :board, :player_1, :player_2, :wargames_flag, :wargames_counter, :wargames_one_wins, :wargames_two_wins
+
+  WIN_COMBINATIONS = [  # win combinations
+  [0,1,2], # Top row
+  [3,4,5], # Middle row
+  [6,7,8], # Bottom row
+  [0,3,6], # left c
+  [1,4,7], # middle c
+  [2,5,8], # right c
+  [0,4,8], # diag lr
+  [2,4,6]  # diag rl
+  ]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new, wargames_flag = false)
+    @board = board
+    @player_1 = player_1
+    @player_2 = player_2
+    @board.token_1 = @player_1.token
+    @board.token_2 = @player_2.token
+    @wargames_flag = wargames_flag
+    @wargames_counter = 0
+    @wargames_one_wins = []
+    @wargames_two_wins = []
+    player_1.game = self
+    player_2.game = self
+    @player_1.turn_value = 1
+    @player_2.turn_value = 2
+    if player_1.class == Players::Computer
+      @player_1.skill = Options.cpu_one_skill[0]
+      if @player_1.skill == 2
+        @player_1.seed = rand(1..100)
+      end
+    end
+    if player_2.class == Players::Computer
+      @player_2.skill = Options.cpu_two_skill[0]
+      if @player_2.skill == 2
+        @player_2.seed = rand(1..100)
+      end
+    end
+  #  self.player_skill_call #troubleshooting
+  end
+
+  def current_player #x goes first
+    board.turn_count.even? ? player_1 : player_2
+  end
+
+  def full?  # if all slots on board are filled by a player_token, return true
+    @board.cells.all? {|i| i == player_1.token || i == player_2.token }
+  end
+
+  def draw?
+    full? && (!!won? == false)
+  end
+
+  def won?
+    Game::WIN_COMBINATIONS.detect do |c|
+      c.all? {|i| board.cells[i] == player_1.token} || c.all? {|i| board.cells[i] == player_2.token}
+    end
+  end
+
+  def winner
+    board.cells[won?[0]] unless won? == nil
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns a truthy value, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.